Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -0,0 +1,691 @@
+                $storageAccounts += $storageAccount
+            }
+            catch
+            {
+                Write-Host "Error fetching Storage Account: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping this Storage Account..." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+     
+    $totalStorageAccount = ($storageAccounts | Measure-Object).Count
+
+    if($totalStorageAccount -eq 0)
+    {  
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "Unable to fetch Storage Account or no Storage Account available." -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+    $folderPath = [Environment]::GetFolderPath("LocalApplicationData") 
+
+    if (Test-Path -Path $folderPath)
+    {  
+        #creating a folder to store the csv file.
+        $folderPath += "\AzTS\Remediation\Subscriptions\$($subscriptionid.replace("-","_"))\$((Get-Date).ToString('yyyyMMdd_hhmm'))\EnableSecureTransit"
+        New-Item -ItemType Directory -Path $folderPath | Out-Null
+    }
+
+    Write-Host "Total Storage Account found: [$($totalStorageAccount)] " -ForegroundColor $([Constants]::MessageType.update)
+    Write-Host $([Constants]::DoubleDashLine)
+
+    write-host "[Step 3 of 6]: Removing the excluded resource group and resource from the list of subscription." 
+    $resourceSummary = @()
+ 
+    # Adding property 'ResourceName' which will contain Storage Account name and being used by common helper method
+    #load the script in the helper.ps file in the Remediation Script directory.. to run -ExcludeResourceGroupNames and -ExcludeResourceNames Parameter .
+    if(-not [string]::IsNullOrWhiteSpace($ExcludeResourceNames) -or -not [string]::IsNullOrWhiteSpace($ExcludeResourceGroupNames))
+    {
+        $storageAccounts | ForEach-Object {
+            $_ | Add-Member -NotePropertyName ResourceName -NotePropertyValue $_.StorageAccountName -ErrorAction SilentlyContinue
+        }
+        # Apply resource or resource group exclusion logic
+        try
+        {
+            $resourceResolver = [ResourceResolver]::new([string] $excludeResourceNames , [string] $excludeResourceGroupNames);
+            $storageAccounts = $resourceResolver.ApplyResourceFilter([PSObject] $storageAccounts) 
+        }
+        catch
+        {
+            Write-Host "Please load Helper.ps1 file in current PowerShell session before executing the script." -ForegroundColor $([Constants]::MessageType.Error)
+            Break
+        }
+
+        if($resourceResolver.messageToPrint -ne $null)
+        {
+            $resourceSummary += "Excluded resource/resource group summary:`n " 
+            $resourceSummary += $resourceResolver.messageToPrint
+        }   
+    }
+    
+    Write-Host $resourceSummary
```

Comment:
`resourceSummary` - If no filter is specified, this will be empty. Should this be moved to around Line 338?
This and the below line.

Reply:
This line should be added but not the next line just to show that step 4 is working and no resource group or resource is excluded.